infogami/infobase/client.py

```diff
@@ -176,12 +174,12 @@ def request(self, sitename, path, method='GET', data=None):
 
         if web.config.debug:
             b = time.time()
-            print("%.02f (%s):" % (round(b-a, 2), web.ctx.infobase_req_count), response.status, method, _path, _data, file=web.debug)
+            print("%.02f (%s):" % (round(b-a, 2), web.ctx.infobase_req_count), response.status_code, method, _path, _data, file=web.debug)
 
-        if response.status == 200:
-            return response.read()
+        if response.status_code == 200:
+            return response.content
```

Comment:
This code was recently converted from Python 2 (`bytes == str`) to Python 3 (`bytes != str`) so please look at the callers of this function to make sure that they are expecting `bytes` (`response.context`) and not `str` (`response.text`).  This has been our biggest issue with our migration to requests.

Reply:
I took a look at the implementation for `response.read()` and it was returning a `bytes` object and that's why I decided to use `response.content` instead.

Ref: https://github.com/python/cpython/blob/master/Lib/http/client.py#L444

Reply:
Yes.  That is the correct choice _in general_ but in this specific case is the _caller_ of this function expecting bytes.  As I said, we have seen this problem before in this codebase.

Reply:
I am not able to find any usage for the `RemoteConnection` class in which this `request` method is in.

Using the `pyright` language server, following the references of `RemoteConnection`:
1. Leads to [this dictionary](https://github.com/internetarchive/infogami/blob/master/infogami/infobase/client.py#L186) which is storing the reference to that class.
2. The dictionary is being used by [this function](https://github.com/internetarchive/infogami/blob/master/infogami/infobase/client.py#L60) according to the type (`local` or `remote`) parameter.
3. That function is being used by [this](https://github.com/internetarchive/infogami/blob/master/infogami/plugins/api/code.py#L74) and [this](https://github.com/internetarchive/infogami/blob/master/infogami/utils/delegate.py#L22) methods which are passing the same `infogami.config.infobase_parameters` information.
4. `infogami.config.infobase_parameters` is coming from [here](https://github.com/internetarchive/infogami/blob/master/infogami/config.py#L35) and that is passing `local` as the type parameter to the function from point 2.

This basically means that the `RemoteConnection` class never seems to be used, atleast internally.

Reply:
Nice detective work!!!  Is there anything in the Open Library repo that would call into this code?  If not, it might be dead code.

Reply:
Yes, openlibrary is actually using this `RemoteConnection` class at 3 instances (I double checked it using the language server as well):
```console
❯ rg "connect\(.*type='remote'" openlibrary/
openlibrary/catalog/olwrite.py
10:        self.conn = client.connect(type='remote', base_url=host)

openlibrary/catalog/importer/olwrite.py
10:        self.conn = client.connect(type='remote', base_url=host)

openlibrary/plugins/openlibrary/connection.py
562:    return client.connect(type='remote', base_url=config.infobase_server)
```
1. https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/olwrite.py#L10
The outcome for the request is being sent to `json.loads` which takes `bytes` and `str` as the input.

2. https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/importer/olwrite.py#L10
Same as 1

3. https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/connection.py#L562
This is being used mainly in the [`OLConnection`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/connection.py#L569) function which is only used [here](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/code.py#L54) to mutate the dictionary from point 1 in my previous comment: `client._connection_types['ol'] = connection.OLConnection`
I don't really see the usage of `ol` connection type anywhere in the openlibrary codebase: 
```console
❯ rg "connect\(.*type='ol'" openlibrary/
```

This should basically conclude that returning `bytes` or `str` both are valid :)

Reply:
Nice.  It looks like you have made the right choice.